scripts/release.sh

```diff
@@ -102,11 +125,31 @@ if [[ $show_reply =~ ^[Yy]$ ]]; then
 	echo -e "$release_notes\n"
 fi
 
-read -p "Create release? (y/n) " -n 1 -r create
+create_message="Create release"
+if ((draft)); then
+	create_message="Create draft release"
+fi
+if ((dry_run)); then
+	create_message+=" (DRYRUN)"
+fi
+read -p "$create_message? (y/n) " -n 1 -r create
 echo
-if [[ $create =~ ^[Yy]$ ]]; then
-	log "Tagging commit $ref as $new_version..."
-	git tag -a "$new_version" -m "$new_version" "$ref"
-	log "Pushing tag to origin..."
-	git push -u origin "$new_version"
+if ! [[ $create =~ ^[Yy]$ ]]; then
+	exit 0
+fi
+
+args=()
+if ((draft)); then
+	args+=(-F draft=true)
 fi
+if ((dry_run)); then
+	args+=(-F dry_run=true)
+fi
+
+gh workflow run release.yaml \
+	--ref "$branch" \
+	-F increment="$increment" \
+	-F snapshot=false \
+	"${args[@]}"
```

Comment:
What's the reason for supporting both a script and the workflow dispatch. I think we should only support workflow dispatch and remove this script.

Reply:
The script allows you to preview the version bump and changelog, unlike the dispatch. Faster response time. Supporting dispatch is IMO a nice bonus, `release.sh` is the primary method.

Reply:
In that case you should probably watch for the run ID and automatically tail it instead, here's how we did it in a now unused script in coder v1: https://github.com/coder/v1/blob/e0b03f984b966349082410be438cd986bb6d47c9/product/coder/deploy/scripts/full_release.sh#L184

Reply:
That does not address previewing the changelog, running the workflow will give you a 1-5 minute delay before you can see it (and doing the whole dry-run first or splitting into two workflows).

Reply:
Sorry, I meant that I am fine with keeping the release.sh script as the primary method and the way it is, but adding automatic job log tailing to the bottom after the user clicks `Y`.

Reply:
Ah sure, that's doable. I was mostly just lazy and told the user how to do it at the end. 😄